Src/FluentAssertions/Equivalency/ObjectReference.cs

```diff
@@ -56,15 +63,12 @@ private bool IsParentOf(ObjectReference other)
         /// <filterpriority>2</filterpriority>
         public override int GetHashCode()
         {
-            unchecked
-            {
-                return (@object.GetHashCode() * 397) ^ path.GetHashCode();
-            }
+            return System.Runtime.CompilerServices.RuntimeHelpers.GetHashCode(@object);
```

Comment:
Because we want to compare/hash the object reference not value. I'm not sure if this is even used anywhere, it looks like `ObjectReference` is only used in lists.

Reply:
I tried a dig a bit in the git history to get some context:
* `ObjectReference` was added in https://github.com/fluentassertions/fluentassertions/commit/5e6d9ee4 where it had a (to me) sensible `GetHashCode`.
* It was changed in https://github.com/fluentassertions/fluentassertions/commit/02e71b30 from `string propertyPath` to `string[] path`, but `GetHashCode` was not updated.
* probably due to that, https://github.com/fluentassertions/fluentassertions/pull/395 did not work as `ObjectReference`s are now stored in `HashSet`.
* https://github.com/fluentassertions/fluentassertions/pull/399 reverted that PR.